File: 3396-valid-word/3396-valid-word.cpp
class Solution {
public:
    bool isValid(string word) {
        if(word.length()<3) return false;

        bool hasvowel=false;
        bool hascons=false;

        for(char c:word){
            if(!isalnum(c)) return false;
            if(isalpha(c)){
                char words=tolower(c);
                if(words=='a'||words=='e'||words=='i'||words=='o'||words=='u'){
                    hasvowel=true;
                }
                else{
                    hascons=true;
                }
            }
        }
        return (hasvowel && hascons)?1:0;
    }
};

File: 3396-valid-word/3396-valid-word.java
class Solution {
    public boolean isValid(String word) {
        if(word.length()<3) return false;

        boolean hasVowel=false;
        boolean hasCons=false;

        for(int i=0;i<word.length();i++){
            if(!Character.isLetterOrDigit(word.charAt(i))){
                return false;
            }
            if(Character.isLetter(word.charAt(i))){
                char ch = Character.toLowerCase(word.charAt(i));
                if(ch=='a'|| ch=='e'|| ch=='i'||ch=='o'||ch=='u'){
                    hasVowel=true;
                }
                else{
                    hasCons=true;
                }
            }
        }

        return (hasVowel && hasCons)?true: false;
    }
}